### DISCUSSION TITLE
Why does supabase_admin role have so many active connections?

### DISCUSSION THREAD

**Discussion by farzd** [supabase/supabase#27671] [Questions] [answered]
```
select
  *
from
  pg_stat_activity;
```

Why has it spun up so many of the same processes ?

<img width="388" alt="image" src="https://github.com/supabase/supabase/assets/1423413/9f39c93b-b5ab-4ee1-a963-4ce99cac63c3">

this was number was at 16 and now has gone down to 1
<img width="999" alt="image" src="https://github.com/supabase/supabase/assets/1423413/a3726ece-2726-49e7-83a2-28a68facea1c">

**Comment by GaryAustin1** [comment, marked as answer]
Realtime has lots (a dozen or so) of postgres connections when there is any postgres changes connection open to it.

**Reply by GaryAustin1** [reply]
This discusses it with a response from SB.  https://github.com/orgs/supabase/discussions/22484#discussioncomment-9096466

**Comment by RishabhPachori** [comment]
@GaryAustin1 


<img width="1083" alt="image" src="https://github.com/user-attachments/assets/c4c12ba4-f2c2-4586-b4e5-eec8091efcc8" />

Please check the image which i have attached

can you please help me to resolve this issue??

In my case supabase_admin has a lot of connections which is causing max connections limit reached

select
  *
from
  pg_stat_activity where usename = 'supabase-admin';

Result : 
| usename                | application_name                       |
| ---------------------- | -------------------------------------- |
|                        |                                        |
| supabase_admin         | TimescaleDB Background Worker Launcher |
| supabase_admin         | pg_cron scheduler                      |
| supabase_admin         |                                        |
| supabase_admin         | pg_net 0.14.0                          |
| supabase_storage_admin |                                        |
| supabase_admin         | supabase_mt_realtime                   |
| supabase_admin         |                                        |
| authenticator          | PostgREST 12.2.0                       |
| supabase_admin         | supabase_mt_realtime                   |
| supabase_admin         | supabase_mt_realtime                   |
| supabase_admin         | supabase_mt_realtime                   |
| supabase_admin         | supabase_mt_realtime                   |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | cluster_node_supavisor@73342e01069e    |
| supabase_admin         | cluster_node_supavisor@73342e01069e    |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         | supavisor_meta                         |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_admin         |                                        |
| supabase_auth_admin    |                                        |
| supabase_auth_admin    |                                        |
|                        |                                        |
|                        |                                        |
|                        |                                        |

**Reply by RishabhPachori** [reply]
@GaryAustin1 You can see in the above result some of the application name are empty and some of them contains supavisor_meta.

**Reply by GaryAustin1** [reply]
I have no idea what all the connections might be used for beyond the realtime ones I did know about from another discussion.
I assume supavisor_meta is you using supavisor database connection.
Typically on an idle instance I see 1/2 dozen or so blank supabase_admin connections.

**Reply by RishabhPachori** [reply]
@GaryAustin1 Can you suggest me something ?? So i can resolve this issue. I am facing this issue in two projects production environment

**Reply by GaryAustin1** [reply]
No.  I got into this one based on realtime stuff as I mentioned.
Are you self-hosted?  If so maybe an issue with your supavisor setup, but I know little about self-hosting.

**Reply by RishabhPachori** [reply]
@GaryAustin1 , Yes i self host supabase
I am on this supavisor version right now

supavisor:
container_name: supabase-pooler
image: supabase/supavisor:1.1.56

What could be the issue ??

**Reply by GaryAustin1** [reply]
As suggested in the discord thread you were in, upgrade your supavisor or overall software.
![image](https://github.com/user-attachments/assets/8ef5580d-aed0-4780-b42a-a032648e76f5)
There have been issues thru time that might relate to supavisor leaving connections open.

**Reply by RishabhPachori** [reply]
Okay @GaryAustin1 i will upgrade the system and check it again

**Reply by RishabhPachori** [reply]
Also @GaryAustin1 I have one project on supabase cloud as well. 
I also checked there as well supabase-admin has 23+ connections and currently i did not call any Database queries but still showing that many connections with supabase-admin

SQL command
select
*
from
pg_stat_activity

output
| usename                | application_name                       |
| ---------------------- | -------------------------------------- |
| supabase_admin         |                                        |
| null                   |                                        |
| supabase_admin         | pg_cron scheduler                      |
| supabase_admin         |                                        |
| supabase_admin         | TimescaleDB Background Worker Launcher |
| supabase_admin         |                                        |
| authenticator          | postgrest                              |
| supabase_admin         | postgres_exporter                      |
| authenticator          | postgrest                              |
| authenticator          | postgrest                              |
| authenticator          | postgrest                              |
| authenticator          | postgrest                              |
| pgbouncer              |                                        |
| supabase_storage_admin |                                        |
| authenticator          | postgrest                              |
| authenticator          | postgrest                              |
| postgres               | postgres-meta 0.0.0-automated          |
| supabase_admin         | realtime_subscription_manager          |
| supabase_admin         | realtime_rls                           |
| supabase_admin         | realtime_subscription_checker          |
| supabase_admin         | realtime_subscription_checker          |
| authenticator          | postgrest                              |
| supabase_admin         | realtime_subscription_manager          |
| supabase_admin         | realtime_subscription_manager_pub      |
| supabase_admin         | realtime_subscription_manager_pub      |
| supabase_admin         | realtime_subscription_manager          |
| supabase_admin         | realtime_subscription_checker          |
| supabase_admin         | realtime_subscription_checker          |
| supabase_admin         | realtime_subscription_manager_pub      |
| supabase_admin         | realtime_subscription_manager          |
| supabase_admin         | realtime_subscription_manager_pub      |
| null                   |                                        |
| null                   |                                        |
| null                   |                                        |


<img width="1440" alt="image" src="https://github.com/user-attachments/assets/163563aa-e670-4c49-afbc-70f08174b0e4" />

**Reply by GaryAustin1** [reply]
The hosted one looks normal.  Realtime takes over a dozen connections.

**Reply by RishabhPachori** [reply]
> The hosted one looks normal. Realtime takes over a dozen connections.

Okay. But some of the entries where usename is null and application_name is empty.
How do we can remove those connections

**Reply by GaryAustin1** [reply]
I don't think you can.  They seem to be normal connections as far as I've seen on my instances.
That number is typical.
If you have want more detail you would need to see if some one from Supabase has more detail for you.
Either support or generated an issue maybe in supabase/supabase issues.

**Reply by RishabhPachori** [reply]
Okay i have created one issue 
https://github.com/orgs/supabase/discussions/36114

**Reply by GaryAustin1** [reply]
You did not create an issue just another discussion which Supabase may or may not see.  There is an issues tab that the devs eventually monitor.

You will very likely not be able to reduce the count beyond what you see in hosted as that appears to be normal.
You might be able to get a answer on what they are used for.
But I would not be waiting around for a "fix" if I were you.
